frontend/apps/pwa/src/components/formFields/types.ts

```diff
@@ -0,0 +1,12 @@
+import { FieldDefinition } from 'core-api-client';
+
+import { FormValue } from '../../reducers/recorder';
+
+export type FieldEditorProps = {
+  field: FieldDefinition;
+  value: string | string[] | null;
+  setValue: (value: string | string[] | null) => void;
```

Comment:
I think a more common naming convention here is `onValueChange`. The component doesn't know that you're "setting" a value, it just knows that you're running something when the value changes.